lib/feature/home/router/home_router_delegate.dart

```diff
@@ -19,14 +20,20 @@ class HomeRouterDelegate extends RouterDelegate
     notifyListeners();
   }
 
+  void setSettingsNavState() {
+    homeNavState = HomeNavState.settings();
+    notifyListeners();
+  }
+
   @override
   Widget build(BuildContext context) {
     return Navigator(
         key: navigatorKey,
         pages: [
           TaskListPage(),
           if (homeNavState is TaskDetailNavState)
-            TaskDetailPage(task: (homeNavState as TaskDetailNavState).task)
+            TaskDetailPage(task: (homeNavState as TaskDetailNavState).task),
+          if (homeNavState is SettingsNavState) SettingsPage()
```

Comment:
Let's change this to instead of replacing the task detail screen, stacking the settings screen on top of whatever home page is displayed and maintaining the order.

Example:
/task_list -> /task_list/settings
/task_list/task_detail -> /task_list/task_detail/settings